src/python/WMCore/WorkQueue/WorkQueueBackend.py

```diff
@@ -321,112 +321,197 @@ def deleteElements(self, *elements):
+        :return: a tuple with the elements accepted and an overview of job counts per site
         """
-        self.logger.info("Getting up to %d available work from %s", numElems, self.queueUrl)
-        self.logger.info("  for team name: %s", team)
-        self.logger.info("  for wfs: %s", wfs)
-        self.logger.info("  with excludeWorkflows: %s", excludeWorkflows)
-        self.logger.info("  for thresholds: %s", thresholds)
-
-        excludeWorkflows = excludeWorkflows or []
+        # NOTE: this method can be less verbose as well
         elements = []
-        sortedElements = []
-
-        # We used to pre-filter sites, looking to see if there are idle job slots
-        # We don't do this anymore, as we may over-allocate
-        # jobs to sites if the new jobs have a higher priority.
-
         # If there are no sites, punt early.
         if not thresholds:
             self.logger.error("No thresholds is set: Please check")
-            return elements, thresholds, siteJobCounts
+            return elements, siteJobCounts
+
+        self.logger.info("Calculating available work from queue %s", self.queueUrl)
 
         options = {}
         options['include_docs'] = True
         options['descending'] = True
-        options['num_elem'] = numElems
         options['resources'] = thresholds
-        if team:
-            options['team'] = team
-        if wfs:
-            result = []
-            for i in xrange(0, len(wfs), 20):
-                options['wfs'] = wfs[i:i + 20]
-                data = self.db.loadList('WorkQueue', 'workRestrictions', 'availableByPriority', options)
-                result.extend(json.loads(data))
-        else:
-            result = self.db.loadList('WorkQueue', 'workRestrictions', 'availableByPriority', options)
-            result = json.loads(result)
-            if not result:
-                self.logger.info("No available work or it did not pass work/data restrictions for: %s ",
-                                 self.queueUrl)
-            else:
-                self.logger.info("Retrieved %d elements from workRestrictions list for: %s",
-                                 len(result), self.queueUrl)
-
-        # Iterate through the results; apply whitelist / blacklist / data
-        # locality restrictions.  Only assign jobs if they are high enough
-        # priority.
-        for i in result:
-            element = CouchWorkQueueElement.fromDocument(self.db, i)
-            # make sure not to acquire work for aborted or force-completed workflows
-            if element['RequestName'] in excludeWorkflows:
-                msg = "Skipping aborted/force-completed workflow: %s, work id: %s"
-                self.logger.info(msg, element['RequestName'], element._id)
-            else:
-                sortedElements.append(element)
-        # sort elements to get them in priority first and timestamp order
+        options['num_elem'] = 9999999  # magic number!
```

Comment:
What is the magic of this number?

Reply:
just a number that we should never reach in the coming 10 (or maybe 100) years or so ;) such that all the elements from workqueue are returned in the same call.